node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -124,12 +140,26 @@ impl ParticipationRequest {
 		let Self { candidate_hash, candidate_receipt, .. } = self;
 		(candidate_hash, candidate_receipt)
 	}
+	// For tests we want to check whether requests are equal, but the
+	// request_timer field of ParticipationRequest doesn't implement
+	// eq. This helper checks whether all other fields are equal,
+	// which is sufficient.
+	#[cfg(test)]
+	pub fn functionally_equal(&self, other: ParticipationRequest) -> bool {
```

Comment:
nit: I had an internal struggle about this, but I think we should just impl `eq` on this type. To me that already indicates functional equality. I read the docs, and there is no restriction mentioned about having to include every field in the equality comparison.

Reply:
Interesting! I hadn't thought of it that way. But it does make sense. I shall make changes.